springfox-swagger-common/src/main/java/springfox/documentation/swagger/csrf/CsrfStrategy.java

```diff
@@ -0,0 +1,208 @@
+/*
+ *
+ *  Copyright 2015 the original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *
+ */
+
+package springfox.documentation.swagger.csrf;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import springfox.documentation.swagger.common.ClassUtils;
+
+/**
+ * The csrf strategy that you are using.
+ * <p>
+ * By default, it follows spring-security's rules of csrf token
+ * creation, storing and matching.
+ * <p>
+ * If you are not using spring-security's csrf support, this
+ * strategy might still be compatible with your own csrf strategy,
+ * but it's not guaranteed, and you need to configure your strategy
+ * using {@link springfox.documentation.swagger.web.UiConfigurationBuilder#csrfStrategy(CsrfStrategy)}
+ *
+ * @author liuxy
+ */
+public class CsrfStrategy {
```

Comment:
Move to `springfox-spring-web`. This doesnt belong on `springfox-swagger-common`